spyder/plugins/ipythonconsole/plugin.py

```diff
@@ -658,13 +661,50 @@ def get_plugin_actions(self):
                                        icon=ima.icon('rename'),
                                        triggered=self.tab_name_editor)
 
-        # Add the action to the 'Consoles' menu on the main window
-        main_consoles_menu = self.main.consoles_menu_actions
-        main_consoles_menu.insert(0, create_client_action)
-        main_consoles_menu += [special_console_menu, connect_to_kernel_action,
-                               MENU_SEPARATOR,
-                               self.interrupt_action, restart_action,
-                               reset_action]
+        # Add actions to the 'Consoles' menu on the main window
+        console_menu = self.main.mainmenu.get_application_menu("consoles_menu")
+        console_menu.aboutToShow.connect(
+                self.update_execution_state_kernel)
+        new_consoles_actions = [
+            create_client_action, special_console_menu,
+            connect_to_kernel_action]
+        restart_connect_consoles_actions = [
+            self.interrupt_action, restart_action, reset_action]
+        for console_new_action in new_consoles_actions:
+            self.main.mainmenu.add_item_to_application_menu(
+                console_new_action,
+                menu_id=ApplicationMenus.Consoles,
+                section=ConsolesMenuSections.New)
+        for console_restart_connect_action in restart_connect_consoles_actions:
+            self.main.mainmenu.add_item_to_application_menu(
+                console_restart_connect_action,
+                menu_id=ApplicationMenus.Consoles,
+                section=ConsolesMenuSections.New)

```

Comment:
Shouldn't this be a different menu section?

Reply:
I mean, you're using `ConsolesMenuSections.New` for the actions above and these ones too.